index.js

```diff
@@ -56,6 +58,19 @@ DuplexWrapper.prototype._write = function _write(input, encoding, done) {
   this._writable.write(input, encoding, done);
 };
 
-DuplexWrapper.prototype._read = function _read(n) {
-  this._readable.resume();
+DuplexWrapper.prototype._read = function _read() {
+  if (this._shouldRead) return;
```

Comment:
I don't even think any of this is necessary anymore with streams3 because listening to the `data` event doesn't switch you into old-mode anymore